src/ekat/physics/constants.yaml

```diff
@@ -0,0 +1,453 @@
+# Base physical constants and their metadata are defined here.
+#
+# EKAT will auto-generate base_constants.* files (where * stands for .hpp, .cpp
+# or .f90) using these values.
+#
+#               ALL CONSTANTS ARE DEFINED IN SI UNITS
+#
+#              time:                seconds   [s]
+#              length:              meters    [m]
+#              mass:                kilograms [kg]
+#              temperature          Kelvin    [K]
+#              pressure             Pascals   [Pa]
+#              force                Newtons   [N]
+#              amount of substance  Moles     [mol]
+#              work                 Joules    [J]
+#
+# Each constant has four required fields:
+#   - software_name: This will be the name of the constant in code (both c++ and f90)
+#   - value: The floating point value (even integers are treated as floating point)
+#   - units: A string containing the units of the constant (must be SI, see above)
+#   - source:  contains the metadata
+#       about the speficic constant's provenance, e.g., where that specific value
+#       to these significant digits is defined.  Allowable fields for source
+#       metadata are:
+#         - name: the name of a recognizable data set, such as NIST CODATA 2018
+#         - citation: a full citation of a reference book or journal article
+#         - url: an internet address (this one MUST be accompanied by additional metadata)
+#         - doi: the DOI of the document that defines the constant
+#         - command: The command used in a software package to output a
+#             constant's numerical value
+#         - definition: Used for a constant whose definition depends on others;
+#             see earth_omega for an example
+#         - note: additional miscellaneous metadata
+#
+#   To change a constant, for example, if NIST updates CODATA 2018 at some point,
+#   the value and the metadata must be changed here.  The updated values
+#   will propagate through the rest of EKAT via its auto-generated constants
+#   modules.
+#
+base_constants:
+  Pi:
+    software_name: pi
```

Comment:
Why not using the software name as the key to the sublist (i.e., the sublist name)? We could add a `description` entry for constants that are not easy to understand (or possibly ambiguous), which would be ignored by the generation script.